tests/test_features/test_pca.py

```diff
@@ -64,39 +65,101 @@ def binary(request):
     request.cls.dataset = Dataset(X, y)
 
 
+@pytest.fixture(scope="class")
+def continuous(request):
+    """
+    Creates a random regressor fixture.
+    """
+    X, y = make_regression(
+        n_samples=500, n_features=22, n_informative=8, random_state=2019
+    )
+
+    # Set a class attribute for continuous data
+    request.cls.continuous = Dataset(X, y)
+
+
 ##########################################################################
 ##PCA Tests
 ##########################################################################
 
 
-@pytest.mark.usefixtures("binary")
+@pytest.mark.usefixtures("binary", "continuous")
 class TestPCADecomposition(VisualTestCase):
     """
     Test the PCADecomposition visualizer
     """
 
-    @pytest.mark.xfail(
-        sys.platform == "win32", reason="images not close on windows (RMSE=4)"
-    )
+    def test_single(self):
+        """
+        Test single target.
+        """
+        visualizer = PCADecomposition(random_state=1998)
+        visualizer.fit(self.continuous.X)
+        visualizer.transform(self.continuous.X)
+        self.assert_images_similar(visualizer)
+
+    def test_continuous(self):
+        """
+        Test continuous target
+        """
+        visualizer = PCADecomposition(colormap="YlOrRd", random_state=2019)
+        assert not hasattr(visualizer, "range_")
+        visualizer.fit(*self.continuous)
+        visualizer.transform(*self.continuous)
+        assert hasattr(visualizer, "range_")
+        visualizer.finalize()
+
+        visualizer.cax.set_yticklabels([])
+
+        self.assert_images_similar(visualizer)
+
+    def test_discrete(self):
+        """
+        Test discrete target.
+        """
+        colors = ["Y", "C0"]
+        classes = ["cats", "dogs"]
+        visualizer = PCADecomposition(colors=colors, classes=classes, random_state=83)
+        assert not hasattr(visualizer, "classes_")
+        visualizer.fit(*self.dataset)
+        assert hasattr(visualizer, "classes_")
+        visualizer.transform(*self.dataset)
+
+        # Make sure that classes are set correctly.
+        npt.assert_array_equal(visualizer.classes_, classes)
+
+        self.assert_images_similar(visualizer)
+
+    def test_fit(self):
+        """
+        Test that fit returns self.
+        """
+        pca = PCADecomposition()
+        assert pca.fit(*self.dataset) is pca
+
+    def test_transform_without_fit(self):
+        """
+        Test that appropriate error is raised when transform called without fit.
+        """
+        oz = PCADecomposition(projection=3)
+        msg = "instance is not fitted yet, please call fit"
+        with pytest.raises(NotFitted, match=msg):
+            oz.transform(*self.continuous)
```

Comment:
Good test! I think the only thing that's missing is transform 2d and 3d, e.g.

```python
@pytest.mark.parametrize("n_components", (2,3))
def test_transform(self, n_components):
    """
    Test transform implements PCA
    """
    Xprime = PCADecomposition(projection=n_components).fit_transform(*self.continous)
    assert Xprime.shape == (500, n_components)
```

Reply:
Yup, I will update this!